src/main/java/step4/InputView.java

```diff
@@ -0,0 +1,55 @@
+package step4;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class InputView {
+    private String carNames;
+    private int moveCount;
+    private List<Car> cars;
+
+    public InputView() {
+    }
+
+    public InputView(String carNames, int moveCount) {
+        this.carNames = carNames;
+        this.moveCount = moveCount;
+        setCars();
+    }
```

Comment:
```suggestion
    public InputView(String carNames, int moveCount) {
        this.moveCount = moveCount;
        this.cars = createUsers(splitCarName(carNames));
    }

    public String[] splitCarName(String carNames) {
        return carNames.split(DILIMITER_COMMA);
    }

    public List<Car> createUsers(String[] splitCarName) {
        return Arrays.stream(splitCarName)
                .map(carName -> new Car(carName, DEFAULT_RACING_RESULT))
                .collect(toList());
    }
```

1. carNames 속성은 불필요 할거 같네요

---

2. setCars() 메서드가 다소 생소하네요. 인자로 아무 값도 받지 않는 부분이나, 생성자를 통해 내부에 할당한 값으로 다시 내부 속성에 값을 세팅한다는 부분이 다소 어색합니다.

---

3. input과 output이 명확해야 테스트하기가 용이합니다. 

---

4. 값을 하드코딩하기보다는 상수로 관리하는 것이 유지보수하기에 좋다고 생각합니다.
상수(static final)를 만들고 이름을 부여하면 이 변수의 역할이 무엇인지 의도를 드러내기에 좋다고 생각합니다.
자바 컨벤션 9.6 Constants를 확인해보세요.

> https://myeonguni.tistory.com/1596

---

5. ArrayList와 같이 구현 클래스가 외부에 노출될 경우, 추후 변경으로 하위호환성이 깨질 우려가 있습니다.

> https://www.slipp.net/questions/345

---

6. Stream API를 활용하여 반복문을 내부 라이브러리에 위임함으로써 무엇을 연산할 것인지 분명히 드러낼 수 있습니다.